contracts/Depository.sol

```diff
@@ -315,9 +376,10 @@ contract Depository is IErrorsTokenomics {
         // Get the LP token address
         address token = product.token;
 
-        // Calculate the payout in OLAS tokens based on the LP pair with the discount factor (DF) calculation
+        // Calculate the payout in OLAS tokens based on the LP pair with the inverse discount factor (IDF) calculation
         // Note that payout cannot be zero since the price LP is non-zero, otherwise the product would not be created
-        payout = IGenericBondCalculator(bondCalculator).calculatePayoutOLAS(tokenAmount, product.priceLP);
+        payout = IBondCalculator(bondCalculator).calculatePayoutOLAS(msg.sender, tokenAmount, product.priceLP,
```

Comment:
Calls new `BondCalculator` contract based on proposed dynamic IDF.

Reply:
What's the dynamic IDF again - how does it differ from previous one?

Reply:
Dynamic IDF based on 4 parameters described in the doc: number of new units per epoch, vesting time (bound by the product vesting time), bond liquidity coefficient (the more one buys from the beginning of the product, the bigger IDF they get), veOLAS amount of the bonder.

In the previous approach the IDF was a pure tokenomics value based on the number of new units added in the epoch, and new owners providing units during the epoch as well.